### DISCUSSION TITLE
Unable to create shared library

### DISCUSSION THREAD

**Discussion by rhythmize** [uNetworking/uWebSockets#968] [Q&A] [answered]
Hi,
I am a bit confused regarding building this repo. Master has no `install` target, doesn't build any library and builds only example applications. But v0.14 branch builds a `shared library` and it's makefile has an `install` target but the branch seems to be diverged from master for almost an year now. 
Moreover the new releases `v.16.*` too doesn't have any `library` building target in the makefile.

How should I proceed to building a shared library and installing it in my system path?

**Comment by rhythmize** [comment]
I need to use this library to implement socket server in my C++ Application.

**Comment by ghost** [comment, marked as answer]
This whole library is header-only, so there is no library to build. You simply drag and drop the src folder to your project or add includepaths. uSockets is the only library you need to build and it has plenty of docs.

**Comment by rhythmize** [comment]
Thanks, for the information.
Moreover, all the modules of this repo are written with structs instead of classes. Is there a specific reason for that?
Because, how I would like to use it is subclassing and providing my custom wrapper to the rest of my application.

**Comment by ghost** [comment]
You have to read the docs and study the examples. There are plenty of info already written.
